src/reflect/scala/reflect/internal/Positions.scala

```diff
@@ -278,12 +279,13 @@ trait Positions extends api.Positions { self: SymbolTable =>
   }
 
   trait PosAssigner extends InternalTraverser {
-    var pos: Position
+    private val _pos: WorkerThreadLocal[Position] = WorkerThreadLocal(NoPosition)
+    @inline def pos: Position = _pos.get
+    @inline def pos_=(position: Position): Unit = _pos.set(position)
   }
   protected[this] lazy val posAssigner: PosAssigner = new DefaultPosAssigner
```

Comment:
why not make this a thread local. BTW it has overrides

Reply:
I thought it will require more changes exactly because of those overrides but when i tried now it turns out it's 3-lines-change. Fixed.